sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -57,124 +49,20 @@ public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpP
-
-                bool shouldRetry = false;
-
-                // We only invoke ShouldRetry for errors. If a user needs full control they can either override HttpPipelinePolicy directly
-                // or modify the ResponseClassifier.
-
-                if (lastException != null || (message.HasResponse && message.Response.IsError))
-                {
-                    shouldRetry = async ? await ShouldRetryAsync(message, lastException).ConfigureAwait(false) : ShouldRetry(message, lastException);
-                }
-
-                if (shouldRetry)
-                {
-                    var retryAfter = message.HasResponse ? message.Response.Headers.RetryAfter : default;
-                    TimeSpan delay = async ? await GetNextDelayAsync(message, retryAfter).ConfigureAwait(false) : GetNextDelay(message, retryAfter);
-                    if (delay > TimeSpan.Zero)
-                    {
-                        if (async)
-                        {
-                            await WaitAsync(delay, message.CancellationToken).ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            Wait(delay, message.CancellationToken);
-                        }
-                    }
-
-                    if (message.HasResponse)
-                    {
-                        // Dispose the content stream to free up a connection if the request has any
-                        message.Response.ContentStream?.Dispose();
-                    }
-
-                    message.RetryNumber++;
-                    AzureCoreEventSource.Singleton.RequestRetrying(message.Request.ClientRequestId, message.RetryNumber, elapsed);
-                    continue;
-                }
-
-                if (lastException != null)
-                {
-                    // Rethrow a singular exception
-                    if (exceptions!.Count == 1)
-                    {
-                        ExceptionDispatchInfo.Capture(lastException).Throw();
-                    }
-
-                    throw new AggregateException(
-                        $"Retry failed after {message.RetryNumber + 1} tries. Retry settings can be adjusted in {nameof(ClientOptions)}.{nameof(ClientOptions.Retry)}" +
-                        $" or by configuring a custom retry policy in {nameof(ClientOptions)}.{nameof(ClientOptions.RetryPolicy)}.",
-                        exceptions);
-                }
-
-                // We are not retrying and the last attempt didn't result in an exception.
-                break;
+            if (async)
+            {
+                await _policy.ProcessAsync(message, processor).ConfigureAwait(false);
+            }
+            else
+            {
+                _policy.Process(message, processor);
             }
-        }
-
-        internal virtual async Task WaitAsync(TimeSpan time, CancellationToken cancellationToken)
```

Comment:
Tests rely on the ability to override the Wait routine.  Is there a way around this?